qt/scientific_interfaces/Inelastic/Manipulation/IElwinView.h

```diff
@@ -0,0 +1,83 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "DllConfig.h"
+#include "IAddWorkspaceDialog.h"
+#include "MantidAPI/MatrixWorkspace_fwd.h"
+#include "MantidQtWidgets/Common/FileFinderWidget.h"
+#include <QModelIndexList>
+#include <QStringList>
+
+#include <memory>
+#include <string>
+#include <tuple>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+class IndirectPlotOptionsView;
+class IElwinPresenter;
+
+class MANTIDQT_INELASTIC_DLL IElwinView {
+
+public:
+  virtual void subscribePresenter(IElwinPresenter *presenter) = 0;
+  virtual void setup() = 0;
+  virtual IndirectPlotOptionsView *getPlotOptions() const = 0;
+  virtual void setFBSuffixes(QStringList const &suffix) = 0;
+
+  virtual void setAvailableSpectra(MantidWidgets::WorkspaceIndex minimum, MantidWidgets::WorkspaceIndex maximum) = 0;
+  virtual void setAvailableSpectra(const std::vector<MantidWidgets::WorkspaceIndex>::const_iterator &from,
+                                   const std::vector<MantidWidgets::WorkspaceIndex>::const_iterator &to) = 0;
+
+  virtual void newPreviewFileSelected(const Mantid::API::MatrixWorkspace_sptr &workspace) = 0;
+  virtual int getCurrentInputIndex() = 0;
+  virtual MantidQt::API::FileFinderWidget *getFileFinderWidget() = 0;
+
+  virtual void plotInput(Mantid::API::MatrixWorkspace_sptr inputWS, int spectrum) = 0;
+  virtual void newInputFiles() = 0;
+  virtual void newInputFilesFromDialog(IDA::IAddWorkspaceDialog const *dialog) = 0;
+  virtual void clearPreviewFile() = 0;
+  virtual void clearInputFiles() = 0;
+  virtual void setRunIsRunning(const bool &running) = 0;
+  virtual void setSaveResultEnabled(const bool &enabled) = 0;
```

Comment:
It is preferred to pass primitive types (such as bool, int, double, etc.) by value rather than by reference. (I think this is a mistake I made for these functions!)

https://stackoverflow.com/questions/14013139/is-it-counter-productive-to-pass-primitive-types-by-reference